src/index.js

```diff
@@ -136,8 +134,12 @@ function syncReduxAndRouter(history, store, selectRouterState = SELECT_STATE) {
        !locationsAreEqual(lastRoute, routing)) {
 
       lastRoute = routing;
-      const method = routing.replace ? 'replaceState' : 'pushState';
-      history[method](routing.state, routing.path);
+
+      const method = routing.replace ? 'replace' : 'push';
+      const locationParam = (!!routing.state && typeof routing.path === 'String') ? 
+        routing.path : { state: routing.state, pathname: routing.path}
```

Comment:
Is there a reason for switching between string and object based on `state`? Much easier to read the code if we can always send an object.